golang/tracereporter.go

```diff
@@ -64,7 +64,7 @@ func NewBinaryAnnotation(key string, value interface{}) BinaryAnnotation {
 
 // TraceReporter is the interface used to report Trace spans.
 type TraceReporter interface {
-	Report(span Span, annotations []Annotation, binaryAnnotations []BinaryAnnotation)
+	Report(span Span, annotations []Annotation, binaryAnnotations []BinaryAnnotation, name string, endpoint *Endpoint)
```

Comment:
We should be much more explicit about what these new parameters are - let's do a separate diff with just the interface change. We need to name / type them better, `name` and `endpoint` are too vague.